### DISCUSSION TITLE
Does nginx-s3-gateway support multiple S3 buckets?

### DISCUSSION THREAD

**Discussion by Nasfy** [nginxinc/nginx-s3-gateway#153] [Q&A]
Hello!
I'm exploring the capabilities of nginx-s3-gateway and I need to know if it can handle requests to multiple S3 buckets simultaneously(different S3_BUCKET_NAME). I would prefer not to set up two proxies.

**Comment by dekobon** [comment]
Hi Nasfy,

The gateway does not currently support multiple buckets in its default configuration. However, you could fairly easily fork the gateway and modify a nginx.conf file such that you are using the same js files for the gateway functionality for two or more buckets.

**Reply by dmrlx** [reply]
Hi @dekobon,

I'm new in nginx and don't really know what exactly needs to be changed in nginx.conf to make it possible to work with multiple buckets. Could you please provide a link to a similar instruction or a search direction. Thanks in advance.

**Reply by dekobon** [reply]
Hi @dmrlx - I do not have any instructions written up nor do I know where you may find instructions. However, if you can find someone well-versed in nginx configuraiton, they should be able to figure out how to modify the provided configuration so that you can use multiple buckets. Another idea is that if you just have two or three buckets, is that you could run two or three instances of the gateway.

I'd love to help, but it would be a lot of work to go through the explanation of how to modify the gateway to work for multiple buckets. Perhaps someone in the community could take this on as a project.

**Reply by dmrlx** [reply]
Hi @dekobon,

Thank you very much! I came up with a scheme with nginx installed on the host and doing proxying requests to docker containers with nginx-s3-gateway, each listening on its own port and redirecting requests to its own bucket. The containers are started using docker-compose. I had to change the nginx configuration for nginx-s3-gateway a bit, but everything works fine now.

**Comment by akashgreninja** [comment]
So this might not exactly be an ideal answer here is what i did 
i shifted from environment to js_val
and added 2 server blocks with the same logic but different js_val
so basically now
https://localhost:80 is one bucket
https://localhost:8080 is another 

i used the standalone script btw
